### PR TITLE
Added support to deployNFT and fixes smart_contract getOptions function

### PR DESCRIPTION
**Pull request by rohan-agarwal-coinbase**
### What changed? Why?
- Adds the ability to call `deployNFT` in order to deploy an ERC721 contract with the appropriate options.
- Also fixed smart_contract.ts `getOptions()` which was not appropriately converting from the model representation to the user-facing representation

### CODE REVIEW

File: src/coinbase/address/wallet_address.ts [hunk 1]
```diff
@@ -378,6 +406,32 @@ export class WalletAddress extends Address {
...
 
+  /**
+   * Creates an ERC721 token contract.
+   *
+   * @param options - The options for creating the ERC721 token.
+   * @param options.name - The name of the ERC721 token.
+   * @param options.symbol - The symbol of the ERC721 token.
+   * @param options.baseURI - The base URI of the ERC721 token.
+   * @returns A Promise that resolves to the deployed SmartContract object.
+   * @throws {Error} If the private key is not loaded when not using server signer.
```
Comment (John-peterson-coinbase) [inline src/coinbase/address/wallet_address.ts:417]: Lets document this may throw an `APIError`

File: src/coinbase/address/wallet_address.ts [hunk 2]
```diff
@@ -351,6 +352,33 @@ export class WalletAddress extends Address {
...
 
+  /**
+   * Deploys an ERC721 token contract.
+   *
+   * @param options - The options for creating the ERC721 token.
+   * @param options.name - The name of the ERC721 token.
+   * @param options.symbol - The symbol of the ERC721 token.
+   * @param options.baseURI - The base URI of the ERC721 token.
+   * @returns A Promise that resolves to the deployed SmartContract object.
+   * @throws {Error} If the private key is not loaded when not using server signer.
```
Comment (John-peterson-coinbase) [inline src/coinbase/address/wallet_address.ts:363]: Let's document this may throw an `APIError`

**Review by John-peterson-coinbase — COMMENTED** [review]
LGTM! 2 small comments on docs.

File: src/coinbase/address/wallet_address.ts (hunk 1 above)
Comment (rohan-agarwal-coinbase) [reply src/coinbase/address/wallet_address.ts:417]: good catch fixed

**Review by John-peterson-coinbase — APPROVED** [review]

**Pull request merged**